src/ui/marker.js

```diff
@@ -177,6 +179,10 @@ export default class Marker extends Evented {
             classList.remove(`mapboxgl-marker-anchor-${key}`);
         }
         classList.add(`mapboxgl-marker-anchor-${this._anchor}`);
+        const classNames = options && options.className ? options.className.trim().split(/\s+/) : [];
+        classNames.forEach((className) => {
+            this._element.classList.add(className);
+        });
```

Comment:
Nit: could likely be simplified:

```suggestion
        classList.add(...classNames);
```

Reply:
Thank you for the suggestion was not sure if "spread syntax" was something commonly used in the lib or not - but good to know it can be used.
Pushed an update in https://github.com/mapbox/mapbox-gl-js/pull/12770/commits/47e2ad1dd37c0de56ff8de57c5f51265318c430b